### ISSUE TITLE
Add surface reconstruction

### ISSUE THREAD

**Issue by chrisfilo** [poldracklab/fmriprep#101]
- Use FreeSurfer 6 beta
- Run FS by default but allow turning it off on the command line
- Make sure to use all available cores for FreeSurfer
- Use `-highres` if the input T1 resolution is <1mm
- Use WM and ventricle parcels produced by FS in aCompCor
- Feed multiple T1s to recon-all if available (but only from one session and only if they have the same dimensions/resolution - for longitudinal datasets each session should be analyzed separately)
- Feed T2s to recon-all if available (but only if they have resolution along the coarsest dimension < 1.2mm)
- Replace internal Freesurfer skullstripping with something better (for example ANTs) - see https://github.com/nipy/nipype/blob/master/nipype/workflows/smri/freesurfer/recon.py#L12
- If T2s are available use sqrt(T1*T2) bias field correction method (see HCP pipelines for details)
- Use FreeSurfer implementation of BBR for EPI to T1 coregistration
- Output: 
  - SUBJECTS_DIR (with all subjects)
  - fMRI timeseries in fsaverage space

Implement and use reportlets outlined in https://github.com/poldracklab/niworkflows/issues/102 and https://github.com/poldracklab/niworkflows/issues/103

**Comment by oesteban** [comment]
Duplicate of #63. But here we have a clear description. I close the old one.

**Comment by effigies** [comment]
Focus group findings
----------
- Verifying white/pial surface alignment with structural was :+1:
  - As a summary stat, noting spikes in cortical thickness (distance from white to pial that is different from surrounding vertices) would be a good addition. (Greg F)
- Interest shown in gifti/cifti outputs (Greg F)
- Mixed response on FreeSurfer 5.3.0 vs 6.0.0
  - Positive response to v6
  - For v5.3, concern was established labs would be more interested in using the same tools, although if they're doing longitudinal, they probably have existing pipelines and aren't going to go back to use FMRIPREP.
  - So our focus should mainly be people setting up new pipelines, so v6.0 is the smarter move

**Comment by effigies** [comment]
Will note here that there weren't any particularly strong opinions about surface sampling techniques (vertex, voxel or pixel based) among the focus group participants I talked to. Not sure if this should be its own issue, given that there's no recommendation to be made.

**Comment by chrisfilo** [comment]
Let's stick with freesurfer defaults and revisit if this comes up again.

**Comment by effigies** [comment]
Proper FreeSurfer v6 support is going to have to wait on resolution of nipy/nipype#1782.

I have a [hacky branch](https://github.com/effigies/nipype/tree/tmp/fs6_balabels) we can pin against just to get it working.

**Comment by chrisfilo** [comment]
Could you work on resolving nipy/nipype#1782? We will need it in a release version of nipype sooner or later.
